integration/run/run_test.go

```diff
@@ -1251,6 +1254,182 @@ func TestCreatingComputeClasses(t *testing.T) {
+	if err != nil {
+		t.Fatal("error creating client for proj2:", err)
+	}
+
+	t.Cleanup(func() {
+		// clean up projects
+		_, err := proj1Client.ProjectDelete(ctx, "proj1")
+		if err != nil {
+			t.Log("failed to delete project 'proj1':", err)
+		}
+		_, err = proj2Client.ProjectDelete(ctx, "proj2")
+		if err != nil {
+			t.Log("failed to delete project 'proj2':", err)
+		}
+	})
+
+	// create both apps, one in proj1 and the other in proj2
+	// app "foo" in proj1 does not publish any ports
+	// app "bar" in proj2 publishes port 80
+	fooImage, err := proj1Client.AcornImageBuild(ctx, "testdata/networkpolicy/Acornfile", nil)
+	if err != nil {
+		t.Fatal("error while building image:", err)
+	}
+	fooApp, err := proj1Client.AppRun(ctx, fooImage.ID, &client.AppRunOptions{
+		Name:            "foo",
+		TargetNamespace: proj1.Namespace,
+	})
+	if err != nil {
+		t.Fatal("error while running app:", err)
+	}
+
+	barImage, err := proj2Client.AcornImageBuild(ctx, "testdata/networkpolicy/publish.Acornfile", nil)
+	if err != nil {
+		t.Fatal("error while building image:", err)
+	}
+	barApp, err := proj2Client.AppRun(ctx, barImage.ID, &client.AppRunOptions{
+		Name:            "bar",
+		TargetNamespace: proj2.Namespace,
+	})
+	if err != nil {
+		t.Fatal("error while running app:", err)
+	}
+
+	// wait for both apps to be ready
+	helper.WaitForObject(t, helper.Watcher(t, proj1Client), &apiv1.AppList{}, fooApp, func(obj *apiv1.App) bool {
+		return obj.Status.Condition(v1.AppInstanceConditionReady).Success
+	})
+	helper.WaitForObject(t, helper.Watcher(t, proj2Client), &apiv1.AppList{}, barApp, func(obj *apiv1.App) bool {
+		return obj.Status.Condition(v1.AppInstanceConditionReady).Success
+	})
+
+	// determine pod IPs so we can test network connections
+	fooIP := getPodIPFromAppName(t, ctx, &kc, fooApp.Name, fooApp.Status.Namespace)
+	barIP := getPodIPFromAppName(t, ctx, &kc, barApp.Name, barApp.Status.Namespace)
+
+	// build an Acorn that just runs a job with the official curl container
+	curlImage1, err := proj1Client.AcornImageBuild(ctx, "testdata/networkpolicy/curl.Acornfile", nil)
+	if err != nil {
+		t.Fatal("error while building image:", err)
+	}
+	curlImage2, err := proj2Client.AcornImageBuild(ctx, "testdata/networkpolicy/curl.Acornfile", nil)
+	if err != nil {
+		t.Fatal("error while building image:", err)
+	}
+
+	checks := []struct {
+		name          string
+		client        client.Client
+		podIP         string
+		imageID       string
+		expectFailure bool
+	}{
+		{
+			name:          "curl-foo-proj1",
+			client:        proj1Client,
+			podIP:         fooIP,
+			imageID:       curlImage1.ID,
+			expectFailure: false,
+		},
+		{
+			name:          "curl-bar-proj1",
+			client:        proj1Client,
+			podIP:         barIP,
+			imageID:       curlImage1.ID,
+			expectFailure: false,
+		},
+		{
+			name:          "curl-foo-proj2",
+			client:        proj2Client,
+			podIP:         fooIP,
+			imageID:       curlImage2.ID,
+			expectFailure: true,
```

Comment:
Just curious, did we do something that makes k3s cluster running in CI being able to implement network policy? I thought the default cni doesn't do network policy.

Reply:
I didn't make any changes to the cluster that runs in CI. k3s uses kube-router's NetPol enforcer (https://docs.k3s.io/networking#network-policy-controller) so I think we should be fine.

Reply:
ah, good to know 👍